File: D1-Morning/PHP/D1-S2-PHP/P4-DataHiding/Circle.php
<?php

class Circle
{
    public $radius;
    private $pi;
    public $area;
    public $perimeter;
    public $diameter;

    public function __construct($radius)
    {
        $this->radius = $radius;
        $this->pi = pi();
    }


    public function getPi(): float
    {
        return $this->pi;
    }


    function getPerimeter(){
        $perimeter = $this->pi * $this->getDiameter();
        return $perimeter;
    }

    function getArea(){
        $area = $this->pi * $this->radius * $this->radius;
        return $area;
    }

    function getDiameter(){
        $diameter = 2 * $this->radius;
        return $diameter;
    }

}

File: D1-Morning/C#/D1-S2/P4-DataHiding/P4-DataHiding/Circle.cs
using System;
namespace CircleExample
{
    public class Circle
    {
        public double radius;
        public const double PI = Math.PI;
        public double area;
        public double perimeter;
        public double diameter;

        public Circle(double radius)
        {
            this.radius = radius;
        }

        public double GetPerimeter()
        {
            perimeter = PI * GetDiameter();
            return perimeter;
        }

        public double GetArea()
        {
            area = PI * radius * radius;
            return area;
        }

        public double GetDiameter()
        {
            diameter = 2 * radius;
            return diameter; 
        }
    }
}
